content/docs/setup/kubernetes/mesh-expansion/index.md

```diff
@@ -5,272 +5,240 @@ weight: 50
+    vm$ sudo chown -R istio-proxy /etc/certs
 
-    {{< text plain >}}
-    $ istio-pilot.istio-system has address 10.150.0.6
     {{< /text >}}
 
-    Check that you can resolve cluster IPs. The actual address will depend on your deployment.
+* Install cluster.env under /var/lib/istio/envoy, owned by istio-proxy.
 
     {{< text bash >}}
-    $ host istio-pilot.istio-system.svc.cluster.local.
-    {{< /text >}}
 
-    Example generated message:
+    vm$ sudo cp cluster.env /var/lib/istio/envoy
+    vm$ sudo chown -R istio-proxy /var/lib/istio/envoy
 
-    {{< text plain >}}
-    istio-pilot.istio-system.svc.cluster.local has address 10.63.247.248
     {{< /text >}}
 
-    Check istio-ingress similarly:
+* Verify the node agent works:
 
     {{< text bash >}}
-    $ host istio-ingress.istio-system.svc.cluster.local.
-    {{< /text >}}
 
-    Example generated message:
+    vm$ sudo node_agent
+    ....
+    CSR is approved successfully. Will renew cert in 1079h59m59.84568493s
 
-    {{< text plain >}}
-    istio-ingress.istio-system.svc.cluster.local has address 10.63.243.30
     {{< /text >}}
 
-*   Verify connectivity by checking whether the VM can connect to Pilot and to an endpoint.
-
-    {{< text bash json >}}
-    $ curl 'http://istio-pilot.istio-system:8080/v1/registration/istio-pilot.istio-system.svc.cluster.local|http-discovery'
-    {
-      "hosts": [
-       {
-        "ip_address": "10.60.1.4",
-        "port": 8080
-       }
-      ]
-    }
+* Start istio using systemctl.
+
+    {{< text bash >}}
+
+    vm$ sudo systemctl start istio-auth-node-agent
+    vm$ sudo systemctl start istio
+
     {{< /text >}}
 
-    On the VM, use the address above. It will directly connect to the pod running istio-pilot.
+After setup, the machine should be able to access services running in the Kubernetes cluster
+or other mesh expansion machines.
+
+Example using /etc/hosts:
 
     {{< text bash >}}
-    $ curl 'http://10.60.1.4:8080/v1/registration/istio-pilot.istio-system.svc.cluster.local|http-discovery'
-    {{< /text >}}
 
-*   Extract the initial Istio authentication secrets and copy them to the machine. The default
-installation of Istio includes Citadel and will generate Istio secrets even if
-the automatic `mTLS`
-setting is disabled (it creates a secret for each service account, and the secret
-is named as `istio.<serviceaccount>`). It is recommended that you perform this
-step to make it easy to enable mutual TLS in the future and to upgrade to a future version
-that will have mutual TLS enabled by default.
+    kubeadm$ kubectl -n bookinfo get svc productpage -o jsonpath='{.spec.clusterIP}'
+10.55.246.247
 
-    `ACCOUNT` defaults to 'default', or `SERVICE_ACCOUNT` environment variable
-    `NAMESPACE` defaults to current namespace, or `SERVICE_NAMESPACE` environment variable
-    (this step is done by machineSetup)
-    On a Mac either `brew install base64` or `set BASE64_DECODE="/usr/bin/base64 -D"`
+    {{< /text >}}
 
```

Comment:
I would take the `# On the VM` and `# On the kubeadm machine` comment out of the code block and add that info on the text before the code block.
For this particular case and after changing the steps to a numbered list, it would look something like this:
1. To obtain the IP address of the cluster run the following command on the \`kubeadm\` machine:
    {{< text bash >}}
    $ kubectl -n bookinfo get svc productpage -o jsonpath='{.spec.clusterIP}'
    10.55.246.247
    {{< /text >}}
1. To add the cluster with the IP address to \`/etc/hosts\` run the following commands on the VM:
    {{< text bash >}}
    $ sudo echo "10.55.246.247 productpage.bookinfo.svc.cluster.local" >> /etc/hosts
    {{< /text >}}
1. To get the HTML content, run the following command on the VM:
    {{< text bash >}}
    $ curl productpage.bookinfo.svc.cluster.local:9080
    {{< /text >}}

Reply:
Just as a suggestion, here's how I did it in the (currently being updated) GCP docs - similar to Rigs' suggestion but I actually bolded which machine was which to avoid any confusion (especially as I occasionally got confused myself when running it):
 https://cloud.google.com/compute/docs/tutorials/istio-on-compute-engine#running_a_service_on_a_mesh_expansion_machine